casper/src/main/scala/coop/rchain/casper/util/comm/CasperPacketHandler.scala

```diff
@@ -0,0 +1,477 @@
+package coop.rchain.casper.util.comm
+
+import cats.data.EitherT
+import cats.effect.concurrent.Ref
+import cats.effect.{Sync, Timer}
+import cats.implicits._
+import cats.{Applicative, Monad}
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.casper._
+import coop.rchain.casper.genesis.Genesis
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.catscontrib.Catscontrib._
+import coop.rchain.catscontrib.{Capture, MonadTrans}
+import coop.rchain.comm.CommError.ErrorHandler
+import coop.rchain.comm.discovery.NodeDiscovery
+import coop.rchain.comm.protocol.rchain.Packet
+import coop.rchain.comm.rp.Connect.RPConfAsk
+import coop.rchain.comm.transport.CommMessages.packet
+import coop.rchain.comm.transport.TransportLayer
+import coop.rchain.comm.{transport, PeerNode}
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.metrics.Metrics
+import coop.rchain.p2p.effects.PacketHandler
+import coop.rchain.shared.{Log, LogSource, Time}
+import monix.eval.Task
+import monix.execution.Scheduler
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
+
+object CasperPacketHandler extends CasperPacketHandlerInstances {
+  private implicit val logSource: LogSource = LogSource(this.getClass)
+
+  def apply[F[_]](implicit ev: CasperPacketHandler[F]): CasperPacketHandler[F] = ev
+
+  def of[
+      F[_]: LastApprovedBlock: Metrics: Timer: BlockStore: NodeDiscovery: TransportLayer: ErrorHandler: RPConfAsk: SafetyOracle: Capture: Sync: Time: Monad: Log: MultiParentCasperRef](
+      conf: CasperConf,
+      delay: FiniteDuration,
+      runtimeManager: RuntimeManager,
+      toTask: F[_] => Task[_])(implicit scheduler: Scheduler): F[CasperPacketHandler[F]] =
```

Comment:
This is pretty hacky 😛

Reply:
It is SUPER hacky but I don't know any other way to do this. I don't know how to implement `Concurrent[EitherT[Task, A, CommError]]`.